src/cogs/when_next.py

```diff
@@ -27,48 +27,44 @@ def __init__(self, bot):
         self.logging = logging.getLogger(__name__)
 
     @commands.command(name="whennext")
-    async def entry(self, context, cmd: str):
+    async def entry(self, context, days: str = "31", weekdays: str = "567", start_offset: str = "0"):
         self.logging.info("whennext called")
-        arg_dict = self._parse_command(cmd)
-        from_date = datetime.date.today()
-        if FROM_KEY in arg_dict:
-            from_date = self._parse_date(context, arg_dict[FROM_KEY])
-            if from_date is False:
-                return
-        until_date = from_date + datetime.timedelta(weeks=2)
-        if UNTIL_KEY in arg_dict:
-            until_date = self._parse_date(context, arg_dict[UNTIL_KEY])
-            if until_date is False:
-                return
-        weekend_days = []
+        days_int = self._parse_int(context, "days", days)
+        if days_int is False:
+            return
+        offset_int = self._parse_int(context, "start_offset", start_offset)
+        if offset_int is False:
+            return
+        from_date = datetime.date.today() + datetime.timedelta(days=offset_int)
+        until_date = from_date + datetime.timedelta(days=days_int)
+        valid_days = []
         while from_date < until_date:
-            if from_date.weekday() > 4:
-                weekend_days.append(from_date.strftime(DATE_FORMAT))
+            if str(from_date.weekday()) in weekdays:
+                valid_days.append(from_date.strftime(DATE_FORMAT))
             from_date = from_date + datetime.timedelta(days=1)
-        ipl = InteractionPipeline(ChoiceInteraction(*weekend_days))
-        await ipl.send_and_watch(
-            context.channel,
-            discord.Embed(
-                title="Hang-out planning",
-                description="Uhhhh so when do you wanna hang out?",
-                colour=discord.Colour.blue(),
-            ),
-            timeout=1,
-        )
+        if valid_days:
+            ipl = InteractionPipeline(ChoiceInteraction(*valid_days))
+            await ipl.send_and_watch(
+                context.channel,
+                discord.Embed(
+                    title="Hang-out planning",
+                    description="Uhhhh so when do you wanna hang out?",
+                    colour=discord.Colour.blue(),
+                ),
+                timeout=1,
+            )
+        else:
+            await context.send(f"Invalid combination of parameters led to no valid days to vote on")
```

Comment:
Can you add a bit more logging so it's possible to work out if something blew up when putting `valid_days` together? Just an `info` or maybe even `debug` before the `for` loop that logs the `from_date` and `until_date` would be useful.

Reply:
Logging added as of 1e786a9f